File: commons/javafx/src/test/java/com/cedarsoft/commons/javafx/NumberStringConverterForIntegerHexTest.java
package com.cedarsoft.commons.javafx;

import static org.assertj.core.api.Assertions.*;

import org.junit.jupiter.api.*;

class NumberStringConverterForIntegerHexTest {

  @Test
  public void name() {
    NumberStringConverterForIntegerHex converter = new NumberStringConverterForIntegerHex();
    assertThat(converter.fromString("-").longValue()).isEqualTo(0);
    assertThat(converter.fromString("-0").longValue()).isEqualTo(0);
    assertThat(converter.fromString("+0").longValue()).isEqualTo(0);
    assertThat(converter.fromString("0").longValue()).isEqualTo(0);
    assertThat(converter.fromString("4").longValue()).isEqualTo(4);
  }
}

File: commons/javafx/src/test/kotlin/it/neckar/open/javafx/NumberStringConverterForIntegerHexTest.kt
package it.neckar.open.javafx

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class NumberStringConverterForIntegerHexTest {
  @Test
  fun name() {
    val converter = NumberStringConverterForIntegerHex()
    assertThat(converter.fromString("-")!!.toLong()).isEqualTo(0)
    assertThat(converter.fromString("-0")!!.toLong()).isEqualTo(0)
    assertThat(converter.fromString("+0")!!.toLong()).isEqualTo(0)
    assertThat(converter.fromString("0")!!.toLong()).isEqualTo(0)
    assertThat(converter.fromString("4")!!.toLong()).isEqualTo(4)
  }
}
